models/llama3/api/datatypes.py

```diff
@@ -206,54 +111,38 @@ class ToolPromptFormat(Enum):
 
 
 @json_schema_type
-class UserMessage(BaseModel):
-    role: Literal[Role.user.value] = Role.user.value
-    content: InterleavedTextMedia
-    context: Optional[InterleavedTextMedia] = None
+class StopReason(Enum):
+    end_of_turn = "end_of_turn"
+    end_of_message = "end_of_message"
+    out_of_tokens = "out_of_tokens"
 
 
-@json_schema_type
-class SystemMessage(BaseModel):
-    role: Literal[Role.system.value] = Role.system.value
-    content: InterleavedTextMedia
+class RawMediaItem(BaseModel):
+    type: Literal["image"] = "image"
+    data: bytes | BytesIO
 
+    model_config = ConfigDict(arbitrary_types_allowed=True)
 
-@json_schema_type
-class ToolResponseMessage(BaseModel):
-    role: Literal[Role.ipython.value] = Role.ipython.value
-    # it was nice to re-use the ToolResponse type, but having all messages
-    # have a `content` type makes things nicer too
-    call_id: str
-    tool_name: Union[BuiltinTool, str]
-    content: InterleavedTextMedia
 
+class RawTextItem(BaseModel):
+    type: Literal["text"] = "text"
+    text: str
 
-@json_schema_type
-class StopReason(Enum):
-    end_of_turn = "end_of_turn"
-    end_of_message = "end_of_message"
-    out_of_tokens = "out_of_tokens"
 
+RawContentItem = Annotated[
+    Union[RawTextItem, RawMediaItem], Field(discriminator="type")
+]
 
-@json_schema_type
-class TokenLogProbs(BaseModel):
-    logprobs_by_token: Dict[str, float]
+RawContent = str | RawContentItem | List[RawContentItem]
 
 
-@json_schema_type
-class CompletionMessage(BaseModel):
-    role: Literal[Role.assistant.value] = Role.assistant.value
-    content: InterleavedTextMedia
-    stop_reason: StopReason
-    tool_calls: List[ToolCall] = Field(default_factory=list)
+class RawMessage(BaseModel):
```

Comment:
This `RawMessage` and `RawContent` types are the most important bits in this PR. Everything is downstream of these changes. See `RawContentItem` as well above.